### ISSUE TITLE
To allow empty content fragment in JSON

### ISSUE THREAD

**Issue by surya057** [adobe/aem-core-wcm-components#720]
## Feature Request

**Is your feature request related to a problem? Please describe.**
Problem Statement:
When an empty content fragment is authored in a page and we try to access model.json of same page, Following error message is thrown "There is no content fragment available to delegate to".

Affected use case:
In our environment, we create editable templates and add content fragments as a default component in our template structure.  Sometimes we need to remove the content fragment reference and leave it empty till we create new content fragments. This in-between period, we need that empty content fragment to be there and we need other components JSON value as well. So in order to achieve this, we need the above feature.

**Describe the solution you'd like**
1) Remove the "throw new IllegalStateException("There is no content fragment available to delegate to");" from ContentFragmentImpl.java.
2) Do Null check to all getter methods in "ContentFragmentImpl.java" class to return type and outside null check, block return an empty value.

Sample Output in JSON will look like below:
contentfragment: {
   description: " ",
   title: " ",
   model: " ",
   :items: { },
       :itemsOrder: [ ],
       elementsOrder: [ ],
    :type: "my_project/components/content/contentfragment"
}

**Are there alternatives?**
No 

**Documentation**
If you can, explain how users will be able to use this and possibly write out a version of the docs.
Maybe a screenshot or design?

**Comment by surya057** [comment]
Hi, 
From my perspective have this as a new enhancement will make sense. Please let me know if there is anything I am missing here.

Thank you,
Surya

**Comment by bpauli** [comment]
@surya057 Yes, I think this request makes sense. I created a PR at https://github.com/adobe/aem-core-wcm-components/pull/729 to return an empty Content Fragment in this case.

**Comment by surya057** [comment]
@bpauli Thank you!
